cpp/src/parquet/encoding.cc

```diff
@@ -2708,7 +2708,14 @@ class DeltaLengthByteArrayDecoder : public DecoderImpl,
 
   void SetData(int num_values, const uint8_t* data, int len) override {
     num_values_ = num_values;
-    if (len == 0) return;
+    if (len == 0) {
+      if (num_values > 0) {
+        throw ParquetException(
+            "Delta length byte array page was empty, but expected {num_values} values.");
```

Comment:
Does this print `num_values`? Perhaps we can check the message with `EXPECT_THROW_THAT`

Reply:
I think num_values could greater than 0, let me write a test for that